### PR TITLE
Summer2021 issue#5697

### PR DESCRIPTION
**Pull request by Holdonbei**
What is the purpose of the change.
this change is aimed to add a module named metrics for nacos/2.0.

Brief changelog
add an interface named MetricsRegister and a class named PrometheusMetricsRegister which i add an unit-test for.

Verifying this change
Follow this checklist to help us incorporate your contribution quickly and easily:
[✅] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
[✅] Format the pull request title like [ISSUE #123] Fix UnknownException when host config not exist. Each commit in the pull request should have a meaningful subject line and body.
[✅] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
[✅] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in test module.
[✅] Run mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true to make sure basic checks pass. Run mvn clean install -DskipITs to make sure unit-test pass. Run mvn clean test-compile failsafe:integration-test to make sure integration-test pass.

### CODE REVIEW

File: test/pom.xml [hunk 1]
```diff
@@ -91,7 +91,7 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>nacos-address</artifactId>
         </dependency>
-
```
Comment (haoyann) [inline test/pom.xml:94]: Please revert this.

Comment (Holdonbei) [reply test/pom.xml]: Thank you for your review, I have been fixed it.

File: metrics/src/main/java/com/alibaba/nacos/metrics/register/PrometheusMetricsRegister.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    
+    /**
+     * registry that contains all metrics from nacos.
+     */
+    private final MeterRegistry registry;
+    
+    /**
+     * registry is injected by spring.
+     *
+     * @param registry registry provided by spring-boot-actuator
```
Comment (haoyann) [inline metrics/src/main/java/com/alibaba/nacos/metrics/register/PrometheusMetricsRegister.java:47]: `MeterRegistry ` from `micrometer-registry-prometheus`

File: metrics/src/test/java/com/alibaba/nacos/metrics/register/PrometheusMetricsRegisterTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.concurrent.TimeUnit;
+
+public class PrometheusMetricsRegisterTest {
+    
+    private final MeterRegistry meterRegistry;
+    
+    private final PrometheusMetricsRegister prometheusMetricsRegister;
+    
+    public PrometheusMetricsRegisterTest() {
+        meterRegistry = new SimpleMeterRegistry();
```
Comment (haoyann) [inline metrics/src/test/java/com/alibaba/nacos/metrics/register/PrometheusMetricsRegisterTest.java:35]: Should test `io.micrometer.prometheus.PrometheusMeterRegistry`

**Review by haoyann — APPROVED** [review]
LGTM

**Review by KomachiSion — COMMENTED** [review]
I'm so confused about this PR. 
What's the difference with call `io.micrometer.core` directly?

**Pull request closed without merging**